src/Integration.Vsix.UnitTests/SonarLintDaemon/SonarLintDaemonTests.cs

```diff
@@ -33,65 +32,81 @@ public class SonarLintDaemonTests
             Directory.CreateDirectory(tempPath);
             Directory.CreateDirectory(storagePath);
-            daemon = new VSIX.SonarLintDaemon(settings, logger, VSIX.SonarLintDaemon.daemonVersion, storagePath, tempPath);
+            testableDaemon = new TestableSonarLintDaemon(settings, logger, VSIX.SonarLintDaemon.daemonVersion, storagePath, tempPath);
+            
         }
 
+        [TestCleanup]
+        public void CleanUp()
+        {
+            ForceDeleteDirectory(tempPath);
+            ForceDeleteDirectory(storagePath);
+        }
+
+
         [TestMethod]
         public void Not_Installed()
         {
-            daemon.IsInstalled.Should().BeFalse();
-            daemon.IsRunning.Should().BeFalse();
+            testableDaemon.IsInstalled.Should().BeFalse();
+            testableDaemon.IsRunning.Should().BeFalse();
         }
 
         [TestMethod]
-        [ExpectedException(typeof(InvalidOperationException))]
         public void Run_Without_Install()
         {
-            daemon.Start();
+            Action op = () => testableDaemon.Start();
+
+            op.Should().ThrowExactly<InvalidOperationException>();
         }
 
         [TestMethod]
         public void Stop_Without_Start_Has_No_Effect()
         {
-            daemon.IsRunning.Should().BeFalse(); // Sanity test
-            daemon.Stop();
-            daemon.IsRunning.Should().BeFalse();
+            testableDaemon.IsRunning.Should().BeFalse(); // Sanity test
+            testableDaemon.Stop();
+            testableDaemon.IsRunning.Should().BeFalse();
         }
 
         [TestMethod]
         [Ignore]
         public void Install_Reinstall_Run()
         {
-            daemon.Install();
+            testableDaemon.Install();
             Directory.GetFiles(tempPath).Length.Should().Be(1);
             Directory.GetDirectories(storagePath).Length.Should().Be(1);
-            Assert.IsTrue(daemon.IsInstalled);
-            Assert.IsFalse(daemon.IsRunning);
+            Assert.IsTrue(testableDaemon.IsInstalled);
+            Assert.IsFalse(testableDaemon.IsRunning);
 
-            daemon.Install();
+            testableDaemon.Install();
             Directory.GetFiles(tempPath).Length.Should().Be(1);
             Directory.GetDirectories(storagePath).Length.Should().Be(1);
-            daemon.IsInstalled.Should().BeTrue();
-            daemon.IsRunning.Should().BeFalse();
+            testableDaemon.IsInstalled.Should().BeTrue();
+            testableDaemon.IsRunning.Should().BeFalse();
 
-            daemon.Start();
-            daemon.IsInstalled.Should().BeTrue();
-            daemon.IsRunning.Should().BeTrue();
-            daemon.Stop();
-            daemon.IsRunning.Should().BeFalse();
+            testableDaemon.Start();
+            testableDaemon.IsInstalled.Should().BeTrue();
+            testableDaemon.IsRunning.Should().BeTrue();
+            testableDaemon.Stop();
+            testableDaemon.IsRunning.Should().BeFalse();
```

Comment:
maybe also check that `testableDaemon.IsInstalled.Should().BeTrue();` is still the case after stopping?

Reply:
This test is ignored and has been for a long time. I haven't tried to fix it as part of this PR.

Reply:
I missed that. And you added more test statements for documentation purposes? If the functionality is broken, do we need a ticket for fixing it?

Reply:
The only changes in this test are as a result of renaming `daemon` to `testableDaemon`.

I doubt this test has ever worked. It's also not a unit test, as it's actually attempting to download the daemon from the web and start running it. I've deleted it.